### PR TITLE
MCP-1620 Fetch Claims endpoint and test

### PR DESCRIPTION
**Pull request by ianwolson**
MCP-1620

### What was the problem?

MCP-1620 Fetch Claims Endpoint

### How does this fix it?

Fixes commit errors 

### Jira Tickets

- [MCP-1620](https://amida.atlassian.net/browse/MCP-1620)

## How to test this PR

- Step 1
- Step 2 

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

File: api/src/main/java/gov/va/vro/api/model/ClaimInfo.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import java.util.List;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+@ToString(includeFieldNames = true)
+public class ClaimInfo {
+  public String claimSubmissionId;
```
Comment (dimitri-amida) [inline api/src/main/java/gov/va/vro/api/model/ClaimInfo.java:14]: These should be private

File: service/db/src/main/java/gov/va/vro/service/db/FetchClaimsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@RequiredArgsConstructor
+public class FetchClaimsServiceImpl implements FetchClaimsService {
+
+  private final ClaimRepository claimRepository;
+  private final ClaimMapper mapper;
+
+  @Override
+  public List<Claim> fetchClaims() {
+    List<ClaimEntity> claimList = claimRepository.findAll();
+    List<Claim> claims = new ArrayList<>();
```
Comment (dimitri-amida) [inline service/db/src/main/java/gov/va/vro/service/db/FetchClaimsServiceImpl.java:24]: Could use streams here to turn it into a one-liner

File: controller/src/main/java/gov/va/vro/controller/VroController.java [hunk 3]
```diff
@@ -137,4 +140,41 @@ public ResponseEntity<FullHealthDataAssessmentResponse> postFullHealthAssessment
...
+        List<String> contentionsList = new ArrayList<>();
+        for (String contention : claim.getContentions()) {
+          contentionsList.add(contention);
+          info.setContentions(contentionsList);
+        }
+        claims.add(info);
+      }
+      FetchClaimsResponse response = new FetchClaimsResponse();
+      if (claims.size() < 1)
+        throw new IllegalStateException("No claims found in DB, claims.size() < 1");
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/VroController.java:164]: we should not throw an exception. If there are no claims, return an empty list

File: controller/src/main/java/gov/va/vro/controller/VroController.java [hunk 4]
```diff
@@ -137,4 +140,41 @@ public ResponseEntity<FullHealthDataAssessmentResponse> postFullHealthAssessment
...
   }
+
+  @Override
+  public ResponseEntity<FetchClaimsResponse> fetchClaims() {
+
+    try {
+
+      List<Claim> claimList = fetchClaimsService.fetchClaims();
+      List<ClaimInfo> claims = new ArrayList<>();
+      for (Claim claim : claimList) {
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/VroController.java:151]: That's fine, but you could use a Mapper to do the work

File: controller/src/main/java/gov/va/vro/controller/VroController.java [hunk 5]
```diff
@@ -137,4 +140,41 @@ public ResponseEntity<FullHealthDataAssessmentResponse> postFullHealthAssessment
...
+        }
+        claims.add(info);
+      }
+      FetchClaimsResponse response = new FetchClaimsResponse();
+      if (claims.size() < 1)
+        throw new IllegalStateException("No claims found in DB, claims.size() < 1");
+      response.setClaims(claims);
+      response.setErrorMessage("Success");
+      return new ResponseEntity<>(response, HttpStatus.OK);
+    } catch (IllegalStateException e) {
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/VroController.java:168]: No need to catch this

File: app/src/test/java/gov/va/vro/FetchClaimsTest.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    assertTrue(response.getClaims().get(1).getContentions().contains("code3"));
+    assertTrue(response.getClaims().get(1).getContentions().contains("code4"));
+
+    // Verify Claim3
+    assertEquals(claim3.getClaimSubmissionId(), response.getClaims().get(2).getClaimSubmissionId());
+    assertEquals(claim3.getVeteran().getIcn(), response.getClaims().get(2).getVeteranIcn());
+    assertEquals(2, response.getClaims().get(2).getContentions().size());
+    assertTrue(response.getClaims().get(2).getContentions().contains("code5"));
+    assertTrue(response.getClaims().get(2).getContentions().contains("code6"));
+  }
```
Comment (dimitri-amida) [inline app/src/test/java/gov/va/vro/FetchClaimsTest.java:102]: NIce! 👍

**Review by dimitri-amida — CHANGES_REQUESTED** [review]
Some of my comments are optional. The ones I consider required are:
- Members of ClaimInfo should be private
- Should not throw exception if there are no claims.

Also please note that the branch is out of date.

**Review by dimitri-amida — APPROVED** [review]
lgtm

**Review by anthonyvelazquez — APPROVED** [review]

**Pull request merged**